akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala

```diff
@@ -469,6 +471,8 @@ final class Broadcast[T](private val outputPorts: Int, eagerCancel: Boolean) ext
 
 object Partition {
 
+  case class PartitionOutOfBoundsException(msg:String) extends RuntimeException(msg) with NoStackTrace
```

Comment:
what is not a good idea? the "case" part?
I "took" it from other places in the project. for example:
https://github.com/akka/akka/blob/a12451e007d2d40564860f5a6a1a675ded8824d5/akka-stream/src/main/scala/akka/stream/scaladsl/One2OneBidiFlow.scala#L14